### PR TITLE
Homs custom status

### PR DESCRIPTION
**Pull request by cristinasewell**
## Description
Add custom status print for `XOffsetMirror`, `OffsetMirror`, `KBOMirror`, `FFMirror`.

## Motivation and Context

Tackling #679 
## How Has This Been Tested?

`XOffsetMirro`
```
In [1]: mr1l0_homs
Out[1]:
MR1L0:HOMS (LFE Mirrors)
------
x_up: (MR1L0:HOMS:MMS:XUP)
------
    position: 201.148
    user_setpoint: -201.071 [um]
    description: Main.M3 / M1L0-Xup
------
pitch: (MR1L0:HOMS:MMS:PITCH)
------
    position: 0.024583
    user_setpoint: -0.000758 [urad]
    description: Main.M5 / M1L0-Pitch
    pitch_enc_rms: 0.029404574106026576
```
`OffsetMirror`
```
In [1]: xrt_m3h
Out[1]:
XRT:M3H (PBT)
------
pitch: (MIRR:XRT:M3H)
------
    position: 98.99555206298828
    setpoint: 99.0 [urad]

```
`KBOMirror`
```
In [1]: mr2k4_kbo
Out[1]:
MR2K4:KBO (TMO Mirrors)
------
x_up: (MR2K4:KBO:MMS:X)
------
    position: 0.94904
    user_setpoint: 0.948485 [mm]
    description: Main.M7 / M2K4-X
------
pitch: (MR2K4:KBO:MMS:PITCH)
------
    position: 13977.016064
    user_setpoint: 13977.043564 [urad]
    description: Main.M9 / M2K4-rY
    pitch_enc_rms: 0.04557812908706964
---------
bender_us (MR2K4:KBO:MMS:BEND:US)
---------
    position 12.745091
    user_setpoint: 12.745164747370715 [urad]
    description: Main.M10 / M2K4-BEND-US
    bender_us_enc_rms: 7.620748375771616e-05
---------
bender_ds (MR2K4:KBO:MMS:BEND:DS)
---------
    position: 18.261711
    user_setpoint: 18.261831823359227 [urad]
    description: Main.M11 / M2K4-BEND-DS
    bender_ds_enc_rms: None
```
`FFMirrror` - not sure what PV i can use to test this.
```
In [1]: from pcdsdevices.mirror import FFMirror as ff

In [2]: f = ff(name='name', prefix='FEE1:M1H')

In [3]: f
Out[3]:
FEE1:M1H
------
x_up: (FEE1:M1H:MMS:X)
------
    position: ERROR
    user_setpoint: None [None]
    description: None
------
pitch: (FEE1:M1H:MMS:PITCH)
------
    position: ERROR
    user_setpoint: None [None]
    description: None
    pitch_enc_rms: None
```
## Where Has This Been Documented?

Docstring

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [x] New/changed functions and methods are covered in the test suite where possible
- [ ] Test suite passes locally
- [x] Test suite passes on travis
- [x] Ran docs/pre-release-notes.sh and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

**Comment by cristinasewell** [comment]
I tried to combine and reuse the same code, but all 4 classes have something different.... (if not different print requirement, different components names) so it was just getting messy...

File: pcdsdevices/mirror.py [hunk 1]
```diff
@@ -214,6 +215,53 @@ def removed(self):
...
+            beamline = get_status_value(md, 'beamline')
+            functional_group = get_status_value(md, 'functional_group')
+            name = f'{self.prefix} ({beamline} {functional_group})'
+
+        p_position = get_status_value(status_info, 'pitch', 'position')
+        p_setpoint = get_status_value(status_info, 'pitch',
+                                      'setpoint', 'value')
+        p_units = get_status_value(status_info, 'pitch', 'setpoint',
+                                   'units')
+        p_p_volts = get_status_value(status_info, 'pitch', 'piezo_volts',
```
Comment (cristinasewell) [inline pcdsdevices/mirror.py:251]: Do we need the `piezo_volts` for the `OffsetMirror`?

Comment (ZLLentz) [reply pcdsdevices/mirror.py:251]: I think `piezo_volts` isn't needed here. Some background on the mechanics:

These pitch axes have two motors working together: a course stepper motor for moving long distances and a fine piezo motor for making small adjustments. So the piezo voltage is essentially wrapped up into the encoder readback for the pitch. It's a useful quantity but I think the scientists wanted to cut down on the number of things displayed here.

Comment (cristinasewell) [reply pcdsdevices/mirror.py:251]: thank you

**Comment by cristinasewell** [comment]
One more thing here, should I display the PVs like they have it in example #679 `(:MMS:XUP)` or full PVs like I have them above? I just took the easy route because I did not think it makes a big difference, but I could be wrong

**Comment by ZLLentz** [comment]
I think how it is now is good- it's not clear if the request was for specifically the PV suffix or if it was a mock-up omitting the prefix. I think this is a reasonable interpretation of the request.

**Review by ZLLentz — APPROVED** [review]
I think this looks great

**Pull request merged**